infra/main.tf

```diff
@@ -54,14 +53,14 @@ resource "aws_iam_role_policy_attachment" "iam_role_policy_attachment" {
     "arn:aws:iam::aws:policy/AWSCloudTrail_FullAccess",
     "arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess"
   ])
-  role       = aws_iam_role.iam_role.name
+  role       = aws_iam_role.ddb_lambda_role.name
```

Comment:
What is the reason to use here `.name` and on line 63 `.arn` ?

Reply:
Take a look at the resources definitions listed bellow.
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy_attachment#role
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#role